olcut-core/src/main/java/com/oracle/labs/mlrg/olcut/util/SubprocessConnection.java

```diff
@@ -0,0 +1,296 @@
+package com.oracle.labs.mlrg.olcut.util;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A connection to a sub-process that can be communicated with over stdio. The subprocess
+ * starts implicitly on activity or can be started explicitly. An idle timeout can also
+ * be used to shut the process down when not in use. Subprocesses should be implemented to take
+ * commands or opcodes and return output followed by an empty line as a response. They should also
+ * watch for a "SHUTDOWN" command (the word "SHUTDOWN" on a line by itself) for terminating
+ * the subprocess cleanly.
+ *
+ * Commands invoked by SubprocessConnection are assumed either to be stateless or to persist
+ * state themselves. It is safe to call {@link #shutdown} on this class to cause the subprocess
+ * to terminate, then to issue more commands, causing the subprocess to be restarted. Setting
+ * an idle timeout for the process with {@link #setTimeout} will cause this behavior to happen
+ * if the idle timeout is reached and new commands are sent.
+ */
+public final class SubprocessConnection {
+    public static final Logger logger = Logger.getLogger(SubprocessConnection.class.getName());
+
+    public static final String SHUTDOWN = "SHUTDOWN";
+    public static final String PYTHONUNBUFFERED = "PYTHONUNBUFFERED";
```

Comment:
Javadoc for the public constants?

Reply:
Sure - though the second string I'll resolve by making it private.